core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -771,7 +778,7 @@ private ExprValue exprDayOfMonth(ExprValue date) {
   /**
    * Day of Week implementation for ExprValue.
    *
-   * @param date ExprValue of Date/String type.
+   * @param date ExprValue of Date/Datetime/String type.
```

Comment:
and time and timestamp

Reply:
Added `Timestamp`. Added javadoc comment to `dayOfWeekToday` since this is used instead for the `TIME` type. Done in 8daa4baf8f4d26b229723884196bd1da23668861.